src/features/user/MFV2/ForestProgress/microComponents/GiftPercentageBar.tsx

```diff
@@ -0,0 +1,30 @@
+import styles from '../ForestProgress.module.scss';
+import { BarsProps } from './Bars';
+
+const GiftPercentageBar = ({
+  personalPercentage,
+  giftPercentage,
+  target,
+  gift,
+  personal,
+}: BarsProps) => {
+  const totalAchievment = gift + personal;
+  return (
+    <div
+      style={{
+        width: `${giftPercentage}%`,
+        borderTopRightRadius: `${
+          target > gift && totalAchievment < target ? 0 : 5
+        }px`,
+        borderBottomRightRadius: `${
+          target > gift && totalAchievment < target ? 0 : 5
+        }px`,
+        borderTopLeftRadius: `${personalPercentage === 0 ? 5 : 0}px`,
+        borderBottomLeftRadius: `${personalPercentage === 0 ? 5 : 0}px`,
+      }}
```

Comment:
You don't really need to programatically provide a border radius for `PersonalPercentage` and `GiftPercentage`. Having border-radius on the parent `barContainer` div with`overflow: hidden` will take care of the border radius.